src/cartservice/Program.cs

```diff
@@ -104,55 +112,27 @@ static void Main(string[] args)
                             Console.WriteLine($"Started as process with id {System.Diagnostics.Process.GetCurrentProcess().Id}");
 
                             // Set hostname/ip address
-                            string hostname = options.Host;
-                            if (string.IsNullOrEmpty(hostname))
-                            {
-                                Console.WriteLine($"Reading host address from {CART_SERVICE_ADDRESS} environment variable");
-                                hostname = Environment.GetEnvironmentVariable(CART_SERVICE_ADDRESS);
-                                if (string.IsNullOrEmpty(hostname))
-                                {
-                                    Console.WriteLine($"Environment variable {CART_SERVICE_ADDRESS} was not set. Setting the host to 0.0.0.0");
-                                    hostname = "0.0.0.0";
-                                }
-                            }
+                            string hostname = ReadParameter("host address", options.Host, CART_SERVICE_ADDRESS, p => p, "0.0.0.0");
 
                             // Set the port
-                            int port = options.Port;
-                            if (options.Port <= 0)
-                            {
-                                Console.WriteLine($"Reading cart service port from {CART_SERVICE_PORT} environment variable");
-                                string portStr = Environment.GetEnvironmentVariable(CART_SERVICE_PORT);
-                                if (string.IsNullOrEmpty(portStr))
-                                {
-                                    Console.WriteLine($"{CART_SERVICE_PORT} environment variable was not set. Setting the port to 8080");
-                                    port = 8080;
-                                }
-                                else    
-                                {
-                                    port = int.Parse(portStr);
-                                }
-                            }
+                            int port = ReadParameter("cart service port", options.Port, CART_SERVICE_PORT, int.Parse, 8080);
 
-                            // Set redis cache host (hostname+port)
-                            ICartStore cartStore;
-                            string redis = ReadRedisAddress(options.Redis);
+                            string projectId = ReadParameter("cloud service project id", options.ProjectId, PROJECT_ID, p => p, null);
 
-                            // Redis was specified via command line or environment variable
-                            if (!string.IsNullOrEmpty(redis))
+                            // Initialize Stackdriver Exporter - currently for tracing only
```

Comment:
The comment seems unnecessary, and no need to say "currently for x only" as it'll probably change and we'll forget to update this. Let the code speak for what's happening.